### PR TITLE
DIT-659 Make Publish allow all files regardless of scan status

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/bindingtarifffilestore/repository/MongoIndexCreator.scala [hunk 1]
```diff
@@ -54,7 +53,7 @@ object MongoIndexCreator {
       indexFieldMappings = Seq(("lastUpdated", IndexType.Ascending)),
       isUnique = false,
       name = Some("expiry_Index"),
-      options = bson.BSONDocument("expireAfterSeconds" -> ttl)
+      options = BSONDocument("expireAfterSeconds" -> ttl, "partialFilterExpression" -> BSONDocument("published" -> false))
```
Comment (gokyo) [inline app/uk/gov/hmrc/bindingtarifffilestore/repository/MongoIndexCreator.scala]: why are we using this `partialFilterExpression` ?

Comment (ed0906) [reply app/uk/gov/hmrc/bindingtarifffilestore/repository/MongoIndexCreator.scala]: We dont want the TTL to apply to published files (they need to last forever)

File: app/uk/gov/hmrc/bindingtarifffilestore/service/FileStoreService.scala [hunk 2]
```diff
@@ -57,29 +57,42 @@ class FileStoreService @Inject()(appConfig: AppConfig,
...
       case READY =>
         val result = scanResult.asInstanceOf[SuccessfulScanResult]
-        attachment.copy(url = Some(result.downloadUrl), scanStatus = Some(READY))
+        val update = attachment.copy(url = Some(result.downloadUrl), scanStatus = Some(READY))
+        if (update.published) {
+          for {
+            updated: Option[FileMetadata] <- repository.update(update)
+            published: Option[FileMetadata] <- updated match {
+              case Some(metadata) => publish(metadata)
+              case _ => Future.successful(None)
```
[the remark this answers is not in the document]
Comment (ed0906) [reply app/uk/gov/hmrc/bindingtarifffilestore/service/FileStoreService.scala]: Good point, I'll add a Log in

[the remark this answers is not in the document]
Comment (ed0906) [reply app/uk/gov/hmrc/bindingtarifffilestore/service/FileStoreService.scala]: Done

**Pull request merged**